pkg/api/resource_types.go

```diff
@@ -193,7 +178,29 @@ func EncodeManifest(manifest, deleteOption, updateStrategy map[string]interface{
 		if err != nil {
 			return nil, fmt.Errorf("failed to unmarshal json to updateStrategy: %v", err)
 		}
-		fmt.Println("upStrategy", upStrategy)
+	}
+
+	// default delete option is Foreground
+	delOption := &workv1.DeleteOption{
+		PropagationPolicy: workv1.DeletePropagationPolicyTypeForeground,
+	}
+
+	// set default delete option to Orphan if update strategy is ReadOnly
+	if upStrategy.Type == workv1.UpdateStrategyTypeReadOnly {
+		delOption = &workv1.DeleteOption{
+			PropagationPolicy: workv1.DeletePropagationPolicyTypeOrphan,
+		}
+	}
+
+	if len(deleteOption) != 0 {
```

Comment:
that means even if the update_strategy is readonly, the delete_option still can be configured. the requirement is if the update_strategy is readonly, the delete_option is orphan. and cannot be configured.

Reply:
updated this to force deletio option to `Orphan` when update strategy is `ReadOnly`.